cmd/contributord/contributord.go

```diff
@@ -15,19 +18,81 @@ import (
 )
 
 var (
-	cacheSize int
-	port      string
+	cacheSize   int
+	port        string
+	addr        string
+	ui          bool
+	uiServeFile string
 )
 
+// TODO: clean this up and use the structs here for the actual return data so we
+// don't have to do anything special for returning the information and manipulating it.
+func buildHtmlData(pullRequests contributed.MergedPullRequestInfo) []pageData {
+
+	var pd []pageData
+
+	for owner, prInfo := range pullRequests {
+		// log.Printf("Owner: %s, Info: %+v\n", owner, prInfo)
+
+		data := pageData{
+			Owner:     owner,
+			AvatarURL: prInfo.AvatarURL,
+			Repos:     []Repository{},
+		}
+
+		for k, v := range prInfo.PullRequests {
+			//log.Printf("Repo: %s, Reqs: %+v\n", k, v)
+
+			r := Repository{
+				Name:         k,
+				PullRequests: []PullReq{},
+			}
+
+			for title, url := range v {
+				//log.Printf("Title: %s, URL: %s\n", title, url)
+				req := PullReq{
+					Title: title,
+					URL:   url,
+				}
+				r.PullRequests = append(r.PullRequests, req)
+			}
+
+			data.Repos = append(data.Repos, r)
+		}
+
+		pd = append(pd, data)
+
+	}
+
+	return pd
+
+}
+
+type PullReq struct {
+	Title string
+	URL   string
+}
+type Repository struct {
+	Name         string
+	PullRequests []PullReq
+}
+type pageData struct {
+	Owner     string
+	AvatarURL string
+	Repos     []Repository
+}
```

Comment:
Note to self: This should be more or less what the data should look like when brought out of `pkg/contributed`, we shouldn't need to manipulate it in order to display it.

This comes from a poor design decision on using `map[string]map[string]string` instead of a neat struct in the first place. Granted, this data is relatively nested and requires multiple loops still, but it is much simpler to access as well if this were the case in the very beginning.

TODO: 
- Refactor the `pkg/contributed` to use this sort of value so that the UI does not need any data manipulation.